pkg/multiplexer/headers.go

```diff
@@ -62,3 +63,35 @@ func forwardedForFromRequest(req *http.Request) string {
 
 	return host
 }
+
+type headerResponseWriter struct {
```

Comment:
Would it make sense to put this logic into the response builder since it has the majority of response building logic?

Reply:
This response wrapper has been removed completely in the latest changes.